zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,992 @@
+    }
+
+    pub fn integer_chip_config(&self) -> IntegerConfig {
+        IntegerConfig::new(self.range_config.clone(), self.main_gate_config.clone())
+    }
+}
+
+pub struct AssignedECDSA<F: FieldExt> {
+    pk_x_le: [AssignedValue<F>; 32],
+    pk_y_le: [AssignedValue<F>; 32],
+    msg_hash_le: [AssignedValue<F>; 32],
+}
+
+#[derive(Debug)]
+pub(crate) struct AssignedSignatureVerify<F: FieldExt> {
+    pub(crate) address: AssignedCell<F, F>,
+    pub(crate) msg_hash_rlc: AssignedCell<F, F>,
+}
+
+// Returns assigned constants [256^1, 256^2, .., 256^{n-1}]
+fn assign_pows_256<F: FieldExt>(
+    ctx: &mut RegionCtx<'_, '_, F>,
+    main_gate: &MainGate<F>,
+    n: usize,
+) -> Result<Vec<AssignedValue<F>>, Error> {
+    let mut pows = Vec::new();
+    for i in 1..n {
+        pows.push(main_gate.assign_constant(ctx, F::from(256).pow(&[i as u64, 0, 0, 0]))?);
+    }
+    Ok(pows)
+}
+
+// Return an array of bytes that corresponds to the little endian representation
+// of the integer, adding the constraints to verify the correctness of the
+// conversion (byte range check included).
+fn integer_to_bytes_le<F: FieldExt, W: WrongExt>(
+    ctx: &mut RegionCtx<'_, '_, F>,
+    main_gate: &MainGate<F>,
+    range_chip: &RangeChip<F>,
+    pows_256: &[AssignedValue<F>],
+    int: &AssignedInteger<W, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+) -> Result<[AssignedValue<F>; 32], Error> {
+    let mut int_le = Vec::new();
+    int_le.extend(int.limbs()[0].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[1].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[2].decompose(9, 8).expect("bad decompose"));
+    int_le.extend(int.limbs()[3].decompose(5, 8).expect("bad decompose"));
+    let int_le: Vec<AssignedValue<F>> = int_le
+        .iter()
+        .map(|b| range_chip.range_value(ctx, &UnassignedValue::from(Some(*b)), 8))
+        .try_collect()
+        .map_err(|e| {
+            error!("RangeChip::range_value error: {:?}", e);
+            e
+        })?;
+    let int_le: [AssignedValue<F>; 32] = int_le.try_into().expect("vec to array of size 32");
+    for (j, positions) in [1..9, 1..9, 1..9, 1..5].iter().enumerate() {
+        let mut acc = int_le[j * 9];
+        for i in positions.clone() {
+            let shifted = main_gate.mul(ctx, &int_le[j * 9 + i], &pows_256[i - 1])?;
+            acc = main_gate.add(ctx, &acc, &shifted)?;
+        }
+        main_gate.assert_equal(ctx, &acc, &(&int.limbs()[j]).into())?;
+    }
+    Ok(int_le)
+}
+
+struct ChipsRef<'a, F: FieldExt, const NUMBER_OF_LIMBS: usize, const BIT_LEN_LIMB: usize> {
+    main_gate: &'a MainGate<F>,
+    range_chip: &'a RangeChip<F>,
+    ecc_chip: &'a GeneralEccChip<Secp256k1Affine, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+    scalar_chip: &'a IntegerChip<secp256k1::Fq, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+    ecdsa_chip: &'a EcdsaChip<Secp256k1Affine, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+}
```

Comment:
I'll add some docs.  The reason I added this is because clippy was complaining about functions having too many arguments, so I joined all the chips into a struct haha.

Reply:
Fixed in https://github.com/appliedzkp/zkevm-circuits/pull/484/commits/562fd93c22667499892bfaad43492b2805ea9647